### PR TITLE
Improved README.md

### PR DESCRIPTION
**Pull request by scohen**
Added more detailed reasoning and explanations about claims made in the readme.

### CODE REVIEW

**Comment by scottming** [comment]
Thanks for upgrading the README. Yesterday, I just drew some C4 diagrams. If you want, you can also add them to the architecture section, they may help users and potential contributors better understand the architecture of lexical.

Or, if you think any areas need to be modified, feel free to let me know.

**System level**

image

**Container level**

image

**Review by scottming — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -4,28 +4,70 @@ Lexical is a next-generation language server for the Elixir programming language
...
+  * None of lexical's dependencies will conflict with your project. This means that lexical can make use of dependencies to make developing in it easier without having to "vendor" them. It also means that you can use lexical to work on your project, **even if lexical depends on your project**.
+  * Your project can depend on a different version of elixir and erlang than lexical itself. This means that lexical can make use of the latest versions of elixir and erlang while still supporting projects that run on older versions.
+  * The build environment for your project is only aware of your project, which enables as-you-type compilation and error reporting.
+  * In the future, there is a possibility of having the lexical vm instance control multiple projects
+
+#### Ease of contribution
+
+Lexical has been designed to be easy to contribute to and work on. It features:
+
+  * A consistent data model that represents the Language Server Protocol. No more working with maps and guessing what the keys of a particular message are.
```
Comment (Blond11516) [inline README.md:32]: I know you've voiced your frustration with this aspect of elixir_sense and elixir_ls, and I understand that frustration having contributed to elixir_ls myself, but this feels unnecessary. I'm all in for talking about the data model as an advantage, but I don't think we need to take directed shots at other projects in the space, even if we're not naming them explicitly.

File: README.md [hunk 2]
```diff
@@ -4,28 +4,70 @@ Lexical is a next-generation language server for the Elixir programming language
...
+  * A set of unit tests and test cases that make testing new features easy.
+
+#### Focus on developer productivity
+
+Lexical is also built with an eye on increasing developer productivity, and approaches some common features a little bit
+differently. For example, Lexical's code completion is _context aware_, which means that if you type `alias MyModule.|`
+you will only receive completions for modules and not the names of functions in `MyModule`. This awareness will extend
+to other areas, which means:
+
+  * You won't see completions for random functions and types in strings. In fact, when extended to string literals, completion will only show you completions if you're inside of an interpolation (`"hello there #{na|}'`).
```
Comment (Blond11516) [inline README.md:44]:
```suggestion
  * You won't see completions for random functions and types in strings. In fact, when extended to string literals, Lexical will only show you completions if you're inside of an interpolation (`"hello there #{na|}'`).
```
"completion will only show you completions" reads a little weird imo.

**Review by Blond11516 — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (scottming) [reply README.md:32]: Maybe `A consistent data model that represents the Language Server Protocol.` is enough.

**Comment by scohen** [comment]
@scottming I think we can do some more work on `pages/architecture.md` and include diagrams there.

File: README.md (hunk 1 above)
Comment (scohen) [reply README.md]: Right, that was not my intention, I was just providing contrast. Removed.

**Comment by scottming** [comment]
> @scottming I think we can do more work on pages/architecture.md and include diagrams there.

Ok, I'll open another pr to do this in the next few days.

**Pull request merged**